src/basic_block.rs

```diff
@@ -340,6 +356,7 @@ impl BasicBlock {
     }
 
     /// Removes this `BasicBlock` from its parent `FunctionValue`. Does nothing if it has no parent.
+    /// It returns `Err(())` when it has no parent to remove from, as LLVM assumes it has a parent.
```

Comment:
I would just cut ", as LLVM assumes it has a parent" here. It makes sense that we issue an error when removing a BB from its parent, and it doesn't have a parent. No further explanation is required.